### PR TITLE
Notification related Changes; Toast;

### PR DESCRIPTION
**Pull request by yattias**
- Refactored toast notification so that it can be used from anywhere in the app. Prior to this, it was only scoped to paper upload flow
- Shortened Notification in dropdown. Truncate if beyond 100 characters
-  Added toast to save to reference manager flow.
- Optimistic Save in save to ref manager dialog

https://www.loom.com/share/6cf1fc5569064d72b9226b7a303255e1

### CODE REVIEW

File: components/Notifications/lib/showSaveToRefManagerToast.tsx
Comment (yattias) [inline components/Notifications/lib/showSaveToRefManagerToast.tsx:1]: Specific toast for saving to ref manager. In future, we may be able to refactor and reuse

Comment (manveerxyz) [reply components/Notifications/lib/showSaveToRefManagerToast.tsx:1]: I've started using chakra-ui at Scholar and love the interfaces/composability of their components. Perhaps can draw some inspiration when we do decide to create a generalized component:
- Toast: https://chakra-ui.com/docs/components/toast/usage#usage
- Modal: https://chakra-ui.com/docs/components/modal#usage

**Review by manveerxyz — COMMENTED** [review]
Overall looks good. I don't have many strong opinions on the toast design after looking at it again, just a comment on the transition below.

**Comment by yattias** [comment]
@manveerxyz agree the animation feels odd. I updated to a fade-in/out effect which looks much nicer

**Comment by manveerxyz** [comment]
> @manveerxyz agree the animation feels odd. I updated to a fade-in/out effect which looks much nicer

I like it! Only thing is now it doesn't seem like the toast goes away unless I click it.

**Comment by yattias** [comment]
Good catch. @manveerxyz just pushed a fix for it.

**Pull request merged**